packages/cozy-harvest-lib/src/models/cipherUtils.js

```diff
@@ -161,7 +161,16 @@ const searchForCipher = async (vaultClient, searchOptions) => {
     uri: konnectorURI,
     type: CipherType.Login
   }
-  const sort = [view => view.login.password === password, 'revisionDate']
+  const sort = [
+    view => (view.login.password === password ? 0 : 1),
+    'revisionDate'
+  ]
   const cipher = await vaultClient.getByIdOrSearch(id, search, sort)
-  return cipher || null
+  if (cipher) {
+    const decrypted = await vaultClient.decrypt(cipher)
+    if (decrypted.login.password == password) {
```

Comment:
i think this part might warrant a comment, even with the PR description it took me a while to understand why we seem to do the check twice.
But I still find `getByIdOrSearch `… strange. We could make the code a lot more explicit if it returned more than one result for example. Anyway, not blocking for me but something to keep in mind.

Reply:
> But I still find getByIdOrSearch … strange. 

Yes I completely agree, I am not at ease with the goal and responsibilities of WebVaultClient.